### DISCUSSION TITLE
Association QueryExpressionMethod using method from another class

### DISCUSSION THREAD

**Discussion by darko1979** [linq2db/linq2db#4188] [Q&A] [answered]
Would it be possible to specify a static method from another class in QueryExpressionMethod of Association attribute?
It would help me to reuse the same expression in multiple association.

**Comment by MaceWindu** [comment, marked as answer]
It is not supported, but your method (or property) could just access shared expression:
```cs
MyExpressionMethod() => Queries.MyQuery;
```

**Reply by darko1979** [reply]
I already did that, I was hoping it could be done to avoid redundant lines.
Could it be implemented to work if expression method name is fully qualified?
Or there is limitation that requires that static method must be in the same class?

**Reply by MaceWindu** [reply]
There is no any limitations, it's just not implemented. You can create feature request (or event better - PR), but personally I would prefer type as separate mapping property.

**Reply by MaceWindu** [reply]
Tracked by #4273
